File: API/Features/CheckIn/Dtos/ReservationWriteDto.cs
using System;
using System.Collections.Generic;

namespace API.Features.CheckIn {

    public class ReservationWriteDto {

        public Guid ReservationId { get; set; }
        public string PutAt { get; set; }
        public List<PassengerWriteDto> Passengers { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/reservations/classes/dtos/form/reservation-write-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { PassengerWriteDto } from './passenger-write-dto'

export interface ReservationWriteDto {

    reservationId: Guid
    customerId: number
    destinationId: number
    driverId?: number
    pickupPointId: number
    portId: number
    portAlternateId: number
    shipId?: number
    date: string
    linkTwistId: string
    refNo: string
    ticketNo: string
    email: string
    phones: string
    adults: number
    kids: number
    free: number
    remarks: string
    passengers: PassengerWriteDto[]
    putAt: string

}
